docs/samples/Microsoft.ML.Samples/Dynamic/TensorFlow/TransferLearning.cs

```diff
@@ -0,0 +1,140 @@
+
+namespace Samples.Dynamic
+{
+    public static class TransferLearning
+    {
+        /// <summary>
+        /// Example use of the TensorFlow image model in a ML.NET pipeline.
+        /// </summary>
+        public static void Example()
+        {
+            var sw = new Stopwatch();
+
+            var mlContext = new MLContext();
+            var data = GetTensorData();
+            var idv = mlContext.Data.LoadFromEnumerable(data);
+
+            // Create a ML pipeline.
+            var pipeline = 
+                mlContext.Transforms.Conversion.MapValueToKey(
+                    nameof(TensorData.Label))
+                .Append(mlContext.Model.ImageClassification(
+                    nameof(TensorData.input),
+                    nameof(TensorData.Label), batchSize: 2,
+                        addBatchDimensionInput: true));
+
+            // Run the pipeline and get the transformed values.
+            var estimator = pipeline.Fit(idv);
+            var transformedValues = estimator.Transform(idv);
+
+            // Retrieve model scores.
+            var outScores = mlContext.Data.CreateEnumerable<OutputScores>(
+                transformedValues, reuseRowObject: false);
+
+            // Display scores. (for the sake of brevity we display scores of the
+            // first 3 classes)
+            foreach (var prediction in outScores)
+            {
+                int numClasses = 0;
+                foreach (var classScore in prediction.Scores.Take(2))
+                {
+                    Console.WriteLine(
+                        $"Class #{numClasses++} score = {classScore}");
+                }
+                Console.WriteLine(
+                    $"Predicted Label: {prediction.PredictedLabel}");
+
+
+                Console.WriteLine(new string('-', 10));
+            }
+
+            Console.WriteLine(sw.Elapsed);
+            // Class #0 score = 0.03416626
+            // Class #1 score = 0.9658337
+            // Predicted Label: 1
+            // ----------
+            // Class #0 score = 0.02959618
+            // Class #1 score = 0.9704038
+            // Predicted Label: 1
+            // ----------
+        }
+
+        private const int imageHeight = 224; 
+        private const int imageWidth = 224;
+        private const int numChannels = 3;
+        private const int inputSize = imageHeight * imageWidth * numChannels;
+
+        /// <summary>
+        /// A class to hold sample tensor data. 
+        /// Member name should match the inputs that the model expects (in this
+        /// case, input).
+        /// </summary>
+        public class TensorData
+        {
+            [VectorType(imageHeight, imageWidth, numChannels)]
+            public float[] input { get; set; }
+
+            public Int64 Label { get; set; }
+        }
+
+        /// <summary>
+        /// Method to generate sample test data. Returns 2 sample rows.
+        /// </summary>
+        public static TensorData[] GetTensorData()
```

Comment:
For example code like this I think it'd be much clearer for users to have a real image-set (TensorData) coming from real image files instead of auto-generated sample test data